src/main/java/org/everit/osgi/dev/maven/AbstractEOSGiMojo.java

```diff
@@ -55,12 +76,38 @@
   @Parameter(property = "executedProject")
   protected MavenProject executedProject;
 
+  private GoogleAnalyticsTrackingService googleAnalyticsTrackingService;
+
+  @Parameter(defaultValue = "${mojoExecution}", readonly = true)
+  protected MojoExecution mojo;
+
   /**
    * The Maven project.
    */
   @Parameter(property = "project")
   protected MavenProject project;
 
+  /**
+   * Skip analytics tracking or not. That means send event statistics to Google Analytics or not.
+   * Default value is <code>false</code> that means send statistics.
+   */
+  @Parameter(property = "eosgi.analytics.skip", defaultValue = "false")
+  private boolean skipAnalytics;
+
+  protected abstract void doExecute() throws MojoExecutionException, MojoFailureException;
+
+  @Override
+  public final void execute() throws MojoExecutionException, MojoFailureException {
+    MojoDescriptor mojoDescriptor = mojo.getMojoDescriptor();
+    String goalName = mojoDescriptor.getGoal();
+    long eventId = getGoogleAnalyticsTrackingService().sendEvent(analyticsReferer, goalName);
+
+    doExecute();
```

Comment:
Surround the doExecute() method with a try-finally and cancel the event sending in the finally block.